File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_map<int,int> mp;
        for(auto &i:arr){
            mp[i]++;
        }
        
        for(auto &i:arr){
            if(i==0){
                if(mp[i]>=2){
                    return true;
                }
            }
            else {if(mp.find(2*i)!=mp.end())return true;}
        }
        
        return false;
        
        
    }
};

File: 1346-check-if-n-and-its-double-exist/1346-check-if-n-and-its-double-exist.java
class Solution {
    public boolean checkIfExist(int[] arr) {
        HashMap<Integer,Integer> mp = new HashMap<>();
        for(int i:arr){
            mp.put(i,mp.getOrDefault(i,0)+1);
        }
        
        for(int i:arr){
            if(i==0){
                if(mp.containsKey(i) && mp.get(i)>=2){
                    return true;
                }
            }
            else{
                if(mp.containsKey(2*i)){
                    return true;
                }
            }
        }
        return false;
    }
}